src/overlays/actors/ovl_En_Dns/z_en_dns.c

```diff
@@ -21,63 +56,466 @@ const ActorInit En_Dns_InitVars = {
+        return;
+    }
+    if (this->actor.params == 0xA) {
+        if (CUR_UPG_VALUE(UPG_NUTS) < 2) {
+            func_8002F434(&this->actor, globalCtx, GI_NUT_UPGRADE_30, 130.0f, 100.0f);
+            return;
+        }
+        func_8002F434(&this->actor, globalCtx, GI_NUT_UPGRADE_40, 130.0f, 100.0f);
+        return;
+    }
+    func_8002F434(&this->actor, globalCtx, this->dnsItemEntry->getItemID, 130.0f, 100.0f);
+}
+
+void func_809EFEE8(EnDns* this, GlobalContext* globalCtx) {
+    if ((func_8010BDBC(&globalCtx->msgCtx) == 5) && (func_80106BC8(globalCtx) != 0)) {
+        func_80106CCC(globalCtx);
+        func_809EFDD0(this, globalCtx);
+        this->actionFunc = func_809EFF50;
+    }
+}
+
+void func_809EFF50(EnDns* this, GlobalContext* globalCtx) {
+    if (Actor_HasParent(&this->actor, globalCtx) != 0) {
+        this->actor.parent = NULL;
+        this->actionFunc = func_809EFF98;
+        return;
+    }
+    func_809EFDD0(this, globalCtx);
+}
+
+void func_809EFF98(EnDns* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+
+    if ((player->stateFlags1 & 0x400) != 0) {
+        if ((func_8010BDBC(&globalCtx->msgCtx) == 6) && (func_80106BC8(globalCtx) != 0)) {
+            this->dnsItemEntry->setRupeesAndFlags(this);
+            this->dropCollectible = 1;
+            this->maintainCollider = 0;
+            this->actor.flags &= -2;
+            EnDns_ChangeAnim(this, 1);
+            this->actionFunc = EnDns_SetupBurrow;
+        }
+    } else {
+        this->dnsItemEntry->setRupeesAndFlags(this);
+        this->dropCollectible = 1;
+        this->maintainCollider = 0;
+        this->actor.flags &= -2;
+        EnDns_ChangeAnim(this, 1);
+        this->actionFunc = EnDns_SetupBurrow;
+    }
+}
+
+void func_809F008C(EnDns* this, GlobalContext* globalCtx) {
+    if ((func_8010BDBC(&globalCtx->msgCtx) == 6) && (func_80106BC8(globalCtx) != 0)) {
+        this->maintainCollider = 0;
+        this->actor.flags &= -2;
+        EnDns_ChangeAnim(this, 1);
+        this->actionFunc = EnDns_SetupBurrow;
+    }
+}
+
+void EnDns_SetupBurrow(EnDns* this, GlobalContext* globalCtx) {
+    f32 frameCount = SkelAnime_GetFrameCount(&D_06004404);
+
+    if (frameCount == this->skelAnime.animCurrentFrame) {
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_AKINDONUTS_HIDE);
+        this->actionFunc = EnDns_Burrow;
+        this->standOnGround = 0;
+        this->yInitPos = this->actor.posRot.pos.y;
+    }
+}
+
+void EnDns_Burrow(EnDns* this, GlobalContext* globalCtx) {
+    f32 depth;
+    Vec3f initPos;
+    s32 i;
+
+    depth = this->yInitPos - this->actor.posRot.pos.y;
+    if ((this->dustTimer & 3) == 0) {
+        initPos.x = this->actor.posRot.pos.x;
+        initPos.y = this->yInitPos;
+        initPos.z = this->actor.posRot.pos.z;
+        func_80028990(globalCtx, 20.0f, &initPos);
+    }
+    this->actor.shape.rot.y += 0x2000;
+    if (400.0f < depth) {
+        if (this->dropCollectible) {
```

Comment:
flip if it matches and && them together

Reply:
Flipping works, but && no longer matches